toolkit/tools/pkg/imagecustomizerlib/customizeoverlays_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCustomizeImageOverlays(t *testing.T) {
+	baseImage := checkSkipForCustomizeImage(t, baseImageTypeCoreEfi)
+
+	testTempDir := filepath.Join(tmpDir, "TestCustomizeImageOverlays")
+	buildDir := filepath.Join(testTempDir, "build")
+	outImageFilePath := filepath.Join(testTempDir, "image.raw")
+	configFile := filepath.Join(testDir, "overlays-config.yaml")
+
+	// Customize image.
+	err := CustomizeImageWithConfigFile(buildDir, configFile, baseImage, nil, outImageFilePath, "raw", "", true, false)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Connect to customized image.
+	imageConnection, err := connectToCoreEfiImage(buildDir, outImageFilePath)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer imageConnection.Close()
+
+	verifyOverlays(t, imageConnection.chroot.RootDir())
+}
+
+func verifyOverlays(t *testing.T, rootPath string) {
+	// Verify fstab for Overlays.
+	fstabPath := filepath.Join(rootPath, "/etc/fstab")
+	fstabContents, err := file.Read(fstabPath)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Check for specific overlay configurations in fstab
+	assert.Contains(t, fstabContents,
+		"overlay /etc overlay lowerdir=/sysroot/etc:/sysroot/home,"+
+			"upperdir=/sysroot/var/overlays/etc/upper,workdir=/sysroot/var/overlays/etc/work,"+
+			"x-systemd.requires=/sysroot/var,x-initrd.mount 0 0")
+
+	assert.Contains(t, fstabContents,
+		"overlay /media overlay lowerdir=/media,"+
+			"upperdir=/var/overlays/media/upper,workdir=/var/overlays/media/work 0 0")
+
+	// Verify that overlays are correctly mounted using the mount command.
+	mountOutput, _, err := shell.Execute("mount")
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Check that the overlays are mounted with the correct options
+	assert.Regexp(t,
```

Comment:
Does this work? I don't think there is anything mounting the overlay devices.

The `connectToCoreEfiImage` function manually specifies the partition list to mount.

Reply:
No, it doesn't work, but why? Is it because after the customization the image has not yet been booted once?

Reply:
The test doesn't boot the customized OS. It just opens the image file and reads the files to ensure that everything was written as expected.

The `connectToImage()` function takes a hardcoded list for the partitions to mount. The list is hardcoded to ensure that the partitions match expectations.

At some point, we will create a new test suite that actually boots the customized OS. That test suite will ensure that the changes we made actually do what we want. But those tests will be slower to run and harder to write.

Reply:
Thanks!